acceptance/tests/lookup/merge_strategies.rb

```diff
@@ -0,0 +1,194 @@
+profiles:
+  webserver:
+    apache:
+      httpd:
+        modules:
+          - auth_kerb
+          - authnz_ldap
+          - cgid
+          - php
+          - status
+arrayed_hash:
+  - array1:
+      key1: valB
+      key3: val3
+YAML
+
+    create_sitepp(master, tmp_environment, <<-SITE)
+notify { "hiera_hash: ${hiera_hash ('profiles')['webserver']['apache']['httpd']['modules']}": }
+notify { "lookup1: ${lookup ('profiles')['webserver']['apache']['httpd']['modules']}": }
+notify { "lookup1b: ${lookup ({'name' => 'profiles', 'merge' => 'deep'})['webserver']['apache']['httpd']['modules']}": }
+notify { "hiera_arrayed_hash: ${hiera_array ('arrayed_hash')}": }
+notify { "lookup_arrayed_hash: ${lookup ({'name' => 'arrayed_hash', 'merge' => {'strategy' => 'deep', 'merge_hash_arrays' => true}})}": }
+    SITE
+
+    on(master, "chmod -R 775 #{fq_tmp_environmentpath}")
+    on(master, "chmod -R 775 #{master_confdir}")
+  end
+
+  step "create another environment, hiera5 config and environment data: #{tmp_environment2}" do
+    create_remote_file(master, "#{fq_tmp_environmentpath2}/hiera.yaml", <<-HIERA)
+---
+version: 5
+hierarchy:
+  - name: "%{environment}/host"
+    data_hash: yaml_data
+    path: "hieradata/host.yaml"
+  - name: "%{environment}/profiles"
+    data_hash: yaml_data
+    path: "hieradata/profiles.yaml"
+HIERA
+
+    on(master, "mkdir -p #{fq_tmp_environmentpath2}/hieradata/")
+    create_remote_file(master, "#{fq_tmp_environmentpath2}/hieradata/host.yaml", <<-YAML)
+---
+profiles:
+  webserver:
+    apache:
+      httpd:
+        modules:
+          - mpm_prefork
+          - php
+          - ssl
+arrayed_hash:
+  - array1:
+      key1: val1
+      key2: val2
+lookup_options:
+  'profiles':
+    merge:
+      strategy: deep
+YAML
+
+    create_remote_file(master, "#{fq_tmp_environmentpath2}/hieradata/profiles.yaml", <<-YAML)
+profiles:
+  webserver:
+    apache:
+      httpd:
+        modules:
+          - auth_kerb
+          - authnz_ldap
+          - cgid
+          - php
+          - status
+arrayed_hash:
+  - array1:
+      key1: valB
+      key3: val3
+lookup_options:
+  'profiles':
+    merge:
+      strategy: deep
+YAML
+
+    create_sitepp(master, tmp_environment2, <<-SITE)
+notify { "hiera_hash: ${hiera_hash ('profiles')['webserver']['apache']['httpd']['modules']}": }
+notify { "lookup2: ${lookup ('profiles')['webserver']['apache']['httpd']['modules']}": }
+notify { "lookup2b: ${lookup ({'name' => 'profiles', 'merge' => 'first'})['webserver']['apache']['httpd']['modules']}": }
+notify { "hiera_arrayed_hash: ${hiera_array ('arrayed_hash')}": }
+notify { "lookup_arrayed_hash: ${lookup ({'name' => 'arrayed_hash', 'merge' => {'strategy' => 'deep', 'merge_hash_arrays' => true}})}": }
+    SITE
+
+    on(master, "chmod -R 775 #{fq_tmp_environmentpath2}")
+  end
+
+  with_puppet_running_on(master,{}) do
+    agents.each do |agent|
+      step "agent lookups #{agent.hostname}, hiera3" do
```

Comment:
the one argument i'm thinking of for splitting it in two is that it could be easier to remove the v3 testing when it comes time.  but this is still very easy in this test, and extremely difficult in other tests like lookup/lookup.rb :-\